tests/phpunit/tests/blocks/wpBlockPatternsRegistry.php

```diff
@@ -482,4 +482,72 @@ public function test_is_registered_for_known_pattern() {
 		$result = $this->registry->is_registered( 'test/one' );
 		$this->assertTrue( $result );
 	}
+
+	/**
+	 * Ensures theme patterns are registered on init.
+	 *
+	 * @ticket 59723
+	 *
+	 * @covers ::_register_theme_block_patterns
+	 */
+	public function test_register_theme_block_patterns_on_init() {
+		// This test needs to use access static class properties.
+		$registry = WP_Block_Patterns_Registry::get_instance();
+
+		// Ensure we're using a theme with patterns.
+		switch_theme( 'twentytwentythree' );
+
+		$theme          = wp_get_theme();
+		$theme_patterns = wp_list_pluck( $theme->get_block_patterns(), 'slug' );
+
+		// This helper is fired on the init hook.
+		_register_theme_block_patterns();
+
+		$registered = wp_list_pluck( $registry->get_all_registered(), 'name' );
+
+		// Cleanup patterns registry.
+		foreach ( $theme_patterns as $pattern ) {
+			$registry->unregister( $pattern );
+		}
+
+		foreach ( $theme_patterns as $pattern ) {
+			$this->assertContains( $pattern, $registered, 'Could not confirm theme patterns were registered.' );
+		}
+	}
+
+	/**
+	 * Ensures theme patterns are not registered when no themes are active and valid.
+	 *
+	 * @ticket 59723
+	 *
+	 * @covers ::_register_theme_block_patterns
+	 */
+	public function test_register_theme_block_patterns_on_init_skipped_during_install() {
+		// This test needs to use access static class properties.
+		$registry = WP_Block_Patterns_Registry::get_instance();
+
+		// Ensure we're using a theme with patterns.
+		switch_theme( 'twentytwentythree' );
+
+		$theme          = wp_get_theme();
+		$theme_patterns = wp_list_pluck( $theme->get_block_patterns(), 'slug' );
+
+		/*
+		 * This will short-circuit theme activation.
+		 * @see wp_get_active_and_valid_themes().
+		 */
+		wp_installing( true );
+
+		// This helper is fired on the init hook.
+		_register_theme_block_patterns();
+
+		$registered = wp_list_pluck( $registry->get_all_registered(), 'name' );
+
+		// Cleanup.
+		wp_installing( false );
+
+		foreach ( $theme_patterns as $pattern ) {
+			$this->assertNotContains( $pattern, $registered, 'Theme patterns were were incorrectly registered.' );
```

Comment:
```suggestion
			$this->assertNotContains( $pattern, $registered, 'Theme patterns were were incorrectly registered.' );
```


Sorry, but expanding on the earlier suggestion so it's possible to know which pattern is failing.

Reply:
Ugh, copy pasta error

```suggestion
			$this->assertNotContains( $pattern, $registered, "The theme pattern '{$pattern}' was incorrectly registered." );
```

Reply:
I'd be ok with adding this, but it seems somewhat duplicative, since the failure message for this assertion will look like this:

```
There was 1 failure:

1) Tests_Blocks_wpBlockPattersRegistry::test_register_theme_block_patterns_on_init
Could not confirm theme patterns were registered.
Failed asserting that an array contains 'twentytwentythree/cta'.
```

Reply:
The notes in the guide are to use different messages when a test contains multiple assertions to assist with debugging ([source](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions)).

I'm not sure if there's an option to suppress the assertion and only show the message for failures though.

Reply:
Generally, I think that guidance is meant to distinguish between multiple distinct assertions in one test, rather than running the same assertion in a loop. However, this did motivate me to refactor the tests a bit in order to avoid unnecessarily looping through each pattern in the list and instead, just comparing the intersection of the two lists to see that they are either all included or none are included (respectively). See: [9fc2962](https://github.com/WordPress/wordpress-develop/pull/5568/commits/9fc2962eb62b01a3cab583e743f19b0aff708fec)